src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs

```diff
@@ -144,13 +151,23 @@ await functionMetadata
                 await FileUtility.WriteAsync(dataFilePath, functionMetadata.TestData);
             }
 
+            var updatedFunctionMetadata = (await _hostFunctionMetadataProvider.GetFunctionMetadataAsync(_languageWorkerOptions.CurrentValue.WorkerConfigs, configChanged))
```

Comment:
If we're going to make these changes to this method, I'd like to put a comment // TEMP HACK on this line pointing to a work item that describes the reason for this workaround, and references the work tracking the actual refactoring work (if we have that work item).

Basically ensuring that anyone looking at this code can understand why we're not using IFunctionMetadataManager as we should be